tests/Mantis/ConfigParserTest.php

```diff
@@ -0,0 +1,432 @@
+	 *
+	 * @param $p_text
+	 * @return string
+	 */
+	private function errorMessage( $p_text ) {
+		return "Original input:\n"
+			. ">>>------------------------\n"
+			. $p_text . "\n"
+			. "<<<------------------------\n";
+	}
+	
+	/**
+	 * Adds a new test case to the list
+	 *
+	 * @param string $p_string
+	 */
+	private function addArrayCase( $p_string ) {
+		$this->cases_array[] = $p_string;
+	}
+
+	/**
+	 * Adds a new scalar test case to the list
+	 *
+	 * @param string $p_string Value to check
+	 * @param string $p_type   Expected type
+	 */
+	private function addScalarCase( $p_string, $p_type ) {
+		$this->cases_scalar[$p_string] = $p_type;
+	}
+
+	/**
+	 * Initialize the Scalar type test cases
+	 */
+	private function initScalarTestCases() {
+		# Integer
+		$this->addScalarCase( '1', PHPUnit_Type::TYPE_INT );
+		$this->addScalarCase( " 1\n", PHPUnit_Type::TYPE_INT );
+
+		# Float
+		$this->addScalarCase( '1.1', PHPUnit_Type::TYPE_FLOAT );
+
+		# String
+		$this->addScalarCase( '"1"', PHPUnit_Type::TYPE_STRING );
+
+		# Built-in string literals
+		$this->addScalarCase( 'null', PHPUnit_Type::TYPE_NULL );
+		$this->addScalarCase( 'false', PHPUnit_Type::TYPE_BOOL );
+		$this->addScalarCase( 'true', PHPUnit_Type::TYPE_BOOL );
+
+		# Constants
+		$this->addScalarCase( 'VERSION_ALL', PHPUnit_Type::TYPE_NULL );         # null
+		$this->addScalarCase( 'VERSION_FUTURE', PHPUnit_Type::TYPE_BOOL );      # false
+		$this->addScalarCase( 'VERSION_RELEASED', PHPUnit_Type::TYPE_BOOL );    # true
+		$this->addScalarCase( 'OFF', PHPUnit_Type::TYPE_INT );                  # 0
+		$this->addScalarCase( 'DEVELOPER', PHPUnit_Type::TYPE_INT );            # int
+		$this->addScalarCase( " DEVELOPER\n", PHPUnit_Type::TYPE_INT );
+		$this->addScalarCase( 'MANTIS_VERSION', PHPUnit_Type::TYPE_STRING );    #string
+		$this->addScalarCase( " MANTIS_VERSION\n", PHPUnit_Type::TYPE_STRING );
+	}
+	
+	/**
+	 * Initialize the array test cases list
+	 */
+	private function initArrayTestCases() {
+		/**
+		 * Template for new test cases
+		 * ---------------------------
+		# comment
+		$this->addArrayCase(
+<<<'EOT'
+EOT
+ 		);
+		 * ---------------------------
+ */
+
+		/**
+		 * Simple arrays
+		 */
+
+		# empty
+		$this->addArrayCase( "array( )" );
+
+		# one element
+		$this->addArrayCase( "array( 1 )" );
+
+		# several elements, trailing delimiter
+		$this->addArrayCase( "array( 1, 2, )" );
+
+		# formatted whitespace
+		$this->addArrayCase( "array ( 1, 2, 3 )" );
+
+		# no whitespace
+		$this->addArrayCase( "array(1,2,3)" );
+
+		# arbitrary whitespace
+		$this->addArrayCase( "  array(\n1,\t2  ,    3 )\r  " );
+
+		# mixed types, quotes
+		$this->addArrayCase(
+<<<'EOT'
+array( 1, 'a', "b" )
+EOT
+		);
+
+		# nested quotes
+		$this->addArrayCase(
+<<<'EOT'
+array( '"a""b"""', "'a''b'''" )
+EOT
+		);
+
+		/**
+		 * Associative arrays
+		 */
+
+		# associative
+		$this->addArrayCase( "array( 0 => 'a', 1 => 'b' )" );
```

Comment:
You mean [like this](https://github.com/mantisbt/mantisbt/pull/761/files#diff-19723a564ef6fb2d9e6afda09158a8d5R410) ?